File: multiple/benchmark_scripts/benchmark.py
#!/usr/bin/python
# encoding:utf-8
import sys
import os
import time
from StringIO import StringIO
buffer = StringIO()
if len(sys.argv) == 1 or os.path.exits(sys.argv[1]):
    print "999.999\nERROR SCRIPT NOT FOUND"
    quit()

inputfile = open(sys.argv[1],  'r')
code = inputfile.read()
inputfile.close()

sys.stdout = buffer
start = time.time()
exec code
end = time.time()

sys.stdout = sys.__stdout__
print end-start
print buffer.getvalue()

File: multiple/benchmark_scripts/benchmark.rb
require 'stringio'

def capture_stdout
  out = StringIO.new
  $stdout = out
  yield
  return out
ensure
  $stdout = STDOUT
end

file_path = ARGV[0]
if file_path.nil?
  puts "999.999\nERROR SCRIPT NOT FOUND"
  exit
end
ctx = File.read(file_path, :encoding => Encoding::UTF_8)
@finish = 0
stdout = capture_stdout do
  start = Time.now
  eval ctx
  @finish = Time.now - start
end
puts @finish
puts stdout.string
